google/cloud/spanner/internal/transaction_impl_test.cc

```diff
@@ -134,8 +132,8 @@ ResultSet Client::Read(std::unique_ptr<SessionHolder>& session,
     }
     switch (mode_) {
       case Mode::kReadSucceeds:  // `begin` -> `id`, calls now parallelized
-        session = ::google::cloud::internal::make_unique<SessionHolder>(
-            session_id_, /*deleter=*/nullptr);
+        session = ::google::cloud::internal::make_unique<internal::Session>(
```

Comment:
This is doing a lot of stuff.

It is creating a `std::unique_ptr<internal::Session, std::default_delete<internal::Session>>` and automatically converting it to `SessionHolder` (which is *not* a type alias, there is an implicit conversion here).

As a side effect, `session` no longer returns the session to the pool, it now simply deallocates.  Is that what we want?

If it is, can we put a comment or something, because it is not super easy to follow.